internal/gengapic/genrest_test.go

```diff
@@ -0,0 +1,231 @@
+		&descriptor.FileDescriptorProto{
+			Package:     proto.String("identify"),
+			Service:     []*descriptor.ServiceDescriptorProto{srv},
+			MessageType: []*descriptor.DescriptorProto{msg},
+		},
+	}
+	req := plugin.CodeGeneratorRequest{
+		Parameter: proto.String("go-gapic-package=path;mypackage,transport=rest"),
+		ProtoFile: fds,
+	}
+	g.init(&req)
+
+	return mthd
+}
+
+func TestPathParams(t *testing.T) {
+	var g generator
+	g.apiName = "Awesome Mollusc API"
+	g.imports = map[pbinfo.ImportSpec]bool{}
+	g.opts = &options{transports: []transport{rest}}
+
+	for tstNum, tst := range []struct {
+		url           string
+		allFieldNames []string
+		expected      map[string]*descriptor.FieldDescriptorProto
+	}{
+		// Null case: no params
+		{
+			url:           "/kingdom",
+			allFieldNames: []string{"name", "mass_kg"},
+			expected:      map[string]*descriptor.FieldDescriptorProto{},
+		},
+		// Reasonable case: params are a subset of fields
+		{
+			url:           "/kingdom/{kingdom}/phylum/{phylum}",
+			allFieldNames: []string{"name", "mass_kg", "kingdom", "phylum"},
+			expected: map[string]*descriptor.FieldDescriptorProto{
+				"kingdom": &descriptor.FieldDescriptorProto{
+					Name:   proto.String("kingdom"),
+					Number: idxAddr(2),
+					Type:   typep(descriptor.FieldDescriptorProto_TYPE_INT32),
+				},
+				"phylum": &descriptor.FieldDescriptorProto{
+					Name:   proto.String("phylum"),
+					Number: idxAddr(3),
+					Type:   typep(descriptor.FieldDescriptorProto_TYPE_INT32),
+				},
+			},
+		},
+		// Degenerate case 1: params and fields are disjoint
+		{
+			url:           "/kingdom/{kingdom}",
+			allFieldNames: []string{"name", "mass_kg"},
+			expected:      map[string]*descriptor.FieldDescriptorProto{},
+		},
+		// Degenerate case 2: params and fields intersect but are not subsets
+		{
+			url:           "/kingdom/{kingdom}/phylum/{phylum}",
+			allFieldNames: []string{"name", "mass_kg", "kingdom"},
+			expected: map[string]*descriptor.FieldDescriptorProto{
+				"kingdom": &descriptor.FieldDescriptorProto{
+					Name:   proto.String("kingdom"),
+					Number: idxAddr(2),
+					Type:   typep(descriptor.FieldDescriptorProto_TYPE_INT32),
+				},
+			},
+		},
+		// Degenerate case 3: fields are a subset of params
+		{
+			url:           "/kingdom/{kingdom}/phylum/{phylum}/class/{class}",
+			allFieldNames: []string{"kingdom", "phylum"},
+			expected: map[string]*descriptor.FieldDescriptorProto{
+				"kingdom": &descriptor.FieldDescriptorProto{
+					Name:   proto.String("kingdom"),
+					Number: idxAddr(0),
+					Type:   typep(descriptor.FieldDescriptorProto_TYPE_INT32),
+				},
+				"phylum": &descriptor.FieldDescriptorProto{
+					Name:   proto.String("phylum"),
+					Number: idxAddr(1),
+					Type:   typep(descriptor.FieldDescriptorProto_TYPE_INT32),
+				},
+			},
+		},
+	} {
+		mthd := setupMethod(t, &g, tst.url, tst.allFieldNames)
+
+		actual := g.pathParams(mthd)
+		if !reflect.DeepEqual(actual, tst.expected) {
```

Comment:
Can you try this?

```
if diff := cmp.Diff(actual, tst.expected, cmp.Comparer(proto.Equal)); diff != "" {
  t.Errord("test %d, got(-),want(+):\n%s", tstNum, diff)
}
```

`cmp` is the Google comparer lib, it's used elsewhere in this project.